### PR TITLE
[dex] - initial chart for coreos/dex

### PR DESCRIPTION
**Pull request by sstarcher**
Initial chart for dex based off the work started in #4107

### CODE REVIEW

File: stable/dex/values.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+replicas: 1
+
+
+service:
+  type: ClusterIP
+  port: 5556
+  # externalIPs:
+
+resources:
```
Comment (timstoop) [inline stable/dex/values.yaml:21]: We do not recommend setting default resources unless there are very good reasons to do so.

**Review by timstoop — CHANGES_REQUESTED** [review]

File: stable/dex/values.yaml (hunk 1 above)
Comment (sstarcher) [reply stable/dex/values.yaml:21]: fixed

**Comment by sstarcher** [comment]
@timstoop found some needed changes after digging into running the chart

**Comment by unguiculus** [comment]
Please follow our RBAC best practices:
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

**Comment by sstarcher** [comment]
@unguiculus good catch I believe I fixed all of the rbac issues.

File: stable/dex/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{{ $fullname := include "dex.fullname" . }}
+{{ $tlsBuiltName := printf "%s-tls" $fullname }}
+{{ $tlsSecretName := default $tlsBuiltName .Values.secret.tlsName }}
+apiVersion: extensions/v1beta1
```
Comment (unguiculus) [inline stable/dex/templates/deployment.yaml]: at least `apps/v1beta2`

File: stable/dex/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+kind: Deployment
+metadata:
+  name: {{ template "dex.fullname" . }}
+  labels:
+    app: {{ template "dex.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (unguiculus) [inline stable/dex/templates/deployment.yaml]: `chart: {{ template "dex.chart" . }}`

Apply everywhere.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by sstarcher** [comment]
@unguiculus done thanks for the review

**Review by unguiculus — APPROVED** [review]

**Comment by sstarcher** [comment]
@unguiculus since you have approved can you drop the do-no-merge/hold label

**Comment by unguiculus** [comment]
Sorry, missed that.

**Comment by fhemberger** [comment]
Could you please add a README for the chart? Also explaining your default `values.yaml` choices?
Not sure why this is in "stable" without any kind of documentation.

**Pull request merged**